src/ILLink.Shared/SourceParameterIndex.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace ILLink.Shared
+{
+	/// <summary>
+	/// Used to indicate the index of the parameter in source code (i.e. the same regardless of if there is a `this` or not)
+	/// </summary>
+	public enum SourceParameterIndex
```

Comment:
I find the naming a bit weird - it's not exactly clear what "source" means here. That said - I didn't come up with a better name really. So I guess we should just add a good comment (maybe add couple of examples).